documentation/src/main/asciidoc/reference/search-dsl-predicate.asciidoc

```diff
@@ -784,11 +784,143 @@ but can be <<search-dsl-predicate-common-boost,boosted>>,
+they will have a `title` field matching the text `robot`
+*or* they will have a `description` field matching the text `investigation`.
+====
+
+[[search-dsl-predicate-or-lambda]]
+=== Adding clauses dynamically with the lambda syntax
+
+It is possible to define the `or` predicate inside a lambda expression.
+This is especially useful when clauses need to be added dynamically to the `or` predicate,
+for example based on user input.
+
+.Easily adding clauses dynamically using `.where(...)` and the lambda syntax
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=or-dynamicParameters-root]
+----
+<1> Get a custom object holding the search parameters provided by the user through a web form, for example.0
+<2> Call `.where(Function)`.
+The function, implemented by a lambda expression, will receive a predicate factory, use it to build an `or` predicate,
+invoke the `with(Consumer)` method and return this predicate. The consumer, implemented by a lambda expression,
+will receive a clause collector for the `or` predicate as an argument, and will add clauses to that collector as necessary.
+<3> Inside the lambda, the code is free to use any Java language constructs, such as `if` or `for`,
+to control the addition of clauses.
+In this case, we only add clauses if the relevant parameter was filled in by the user.
+====
+
+Another syntax relying on the method `with(...)` can be useful when the boolean predicate is not the root predicate:
+
+.[[example-and-or-dynamicParameters-with]]Easily adding clauses dynamically using `with(...)` and the lambda syntax
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=and-or-dynamicParameters-with]
+----
+<1> Get a custom object holding the search parameters provided by the user through a web form, for example.
+<2> Call `.where(Function)` to create the root predicate, which is not the one we're interested in here.
+<3> Call `.or().with(Consumer)` to create an inner, non-root predicate.
+The consumer, implemented by a lambda expression, will receive a collector as an argument
+and will add clauses to that collector as necessary.
+<4> Inside the lambda, the code is free to use any Java language constructs, such as `if` or `for`,
+to control the addition of clauses.
+In this case, we add one clause per author filter.
+====
+
+[[search-dsl-predicate-or-other]]
+=== Options
+
+* The score of an `or` predicate is variable by default,
+but can be <<search-dsl-predicate-common-constantScore,made constant with `.constantScore()`>>.
+* The score of an `or` predicate can be <<search-dsl-predicate-common-boost,boosted>>
+with a call to `.boost(...)`.
+
 [[search-dsl-predicate-boolean]]
-== [[_combining_queries]] `bool`: combine predicates (or/and/...)
+== `bool`: advanced combinations of predicates (or/and/...)
```

Comment:
```suggestion
== [[_combining_queries]] `bool`: advanced combinations of predicates (or/and/...)
```

This is a pre-existing tag from Hibernate Search 5, we need to keep it to avoid dead links.